doc/bgp_pic/bgp_pic.md

```diff
@@ -0,0 +1,200 @@
+<!-- omit in toc -->
+# BGP PIC HLD
+<!-- omit in toc -->
+### Revision
+| Rev |     Date    |       Author           | Change Description                |
+|:---:|:-----------:|:----------------------:|-----------------------------------|
+| 0.1 | Oct  8 2023 |   Eddie Ruan / lingyu Zhang   | Initial Draft              |
+
+<!-- omit in toc -->
+## Table of Content
+- [Goal \& Scope](#goal--scope)
+
+## Goal and Scope
+BGP PIC, as detailed in the RFC available at https://datatracker.ietf.org/doc/draft-ietf-rtgwg-bgp-pic/, addresses the enhancement of BGP route convergence. This document outlines a method to arrange forwarding structures that can lead to improved BGP route convergence. BGP PIC offers two primary enhancements:
+
+1. It prevents BGP load balancing updates from being triggered by IGP load balancing updates. This issue, which is discussed in the SONiC Routing Working Group (https://lists.sonicfoundation.dev/g/sonic-wg-routing/files/SRv6%20use%20case%20-%20Routing%20WG.pptx), can be effectively resolved using BGP PIC.
+
+<figure align=center>
+    <img src="images/srv6_igp2bgp.jpg" >
+    <figcaption>Figure 1. Alibaba issue Underlay routes flap affecting Overlay SRv6 routes <figcaption>
+</figure> 
+Note: we only handle VPN overlay routes via BGP PIC. For global table's recursive routes handling, it would be handled via a seperate HLD and done by Accton team.
+
+2. We aim to achieve fast convergence in the event of a hardware forwarding failure related to a remote BGP PE becoming unreachable. Convergence in the slow path forwarding mode is not a priority.
+
+## High Level Design
+One of the challenges in implementing PIC within FRR is the absence of PIC support in the Linux kernel. To minimize alterations in FRR while enabling PIC on platforms that do not require Linux kernel support for this feature, we are primarily focused on two key modifications:
+1. In the 'zebra' component:
+    - Introduce a new Next Hop Group (PIC-NHG) specifically for the FORWARDING function. This NHG will serve as the shareable NHG in hardware.
```

Comment:
This PIC-NHG will have a nexthop group id. How will that new nexthop group id be allocated ? by BGP ?
if we have a PIC-NHG with multipath, so that means we will have 3 nexthop ids, 1 for the parent, and two for each of the nexthops ?

Reply:
I will keep current zebra's approach to allocate NHG id. I don't plan to move it to BGP.  So yes for your first question, BGP will continue sending routes update as usual. 

Yes for your second question as well. That is current zebra's approach.

Reply:
thanks Eddie.